src/main/java/net/glowstone/entity/objects/GlowItemFrame.java

```diff
@@ -196,10 +200,10 @@ public void setFacingDirection(BlockFace blockface) {
 
     @Override
     public void setItem(ItemStack is) {
-        if (is == null) {
-            is = new ItemStack(Material.AIR, 1);
-        }
+        is = InventoryUtil.itemOrEmpty(is);
+        is = is.clone();
         is.setAmount(1);
```

Comment:
Cloning is required, imo.
During testing, I had the following scenario:
- I have 4 Rails in my Inventory
- I put one into the itemframe. Now I have 3 Rails in my Inventory.
- I rightclick onto the itemframe, and the items drop.
- I pick them up, and now I have 6 Rails in my Inventory.
Cloning solved this scenario.

Also, at least on paper, the amount of the itemstack seems to be set to 1.